service/history/workflow_task_handler.go

```diff
@@ -211,6 +212,45 @@ func (handler *workflowTaskHandlerImpl) handleCommands(
 	return mutations, nil
 }
 
+func (handler *workflowTaskHandlerImpl) ensureUpdatesProcessed(
+	_ context.Context,
+	workflowTaskScheduledEventID int64,
+	workflowTaskHeartbeating bool,
+) error {
+
+	// If WT is already about to fail with another cause, don't check for unprocessed updates.
+	if handler.workflowTaskFailedCause != nil {
+		return nil
+	}
+
+	// If WT is a heartbeat WT, then it doesn't have to have messages.
+	if workflowTaskHeartbeating {
+		return nil
+	}
+
+	// If worker has just completed workflow with COMPLETE_WORKFLOW_EXECUTION command, then it doesn't have to have messages.
+	if !handler.mutableState.IsWorkflowExecutionRunning() {
```

Comment:
What if the workflow has continued as new?

Reply:
CAN will complete WF and unprocessed updates will timeout. CAN is not different from normal WF complete here.